nuxeo-services/nuxeo-platform-usermanager-core/src/main/java/org/nuxeo/ecm/platform/usermanager/UserManagerImpl.java

```diff
@@ -599,10 +599,15 @@ protected boolean useCache() {
 
     @Override
     public NuxeoPrincipal getPrincipal(String username) {
-        if (useCache()) {
+        return getPrincipal(username, true);
+    }
+
+    @Override
+    public NuxeoPrincipal getPrincipal(String username, boolean fetchReferences) {
+        if (useCache() && fetchReferences) {
```

Comment:
Does this mean that we don't cache a principal when not fetching its references? Might deserve a comment.

Reply:
yes, we don't cache it: the cached principal would only reference the already cached user model => we agreed with TR that this cache was not worth it.
yes, I could add a comment